### PR TITLE
Raw data and CTF for FDD

### PR DESCRIPTION
**Pull request by mbroz84**
Raw data format, raw reader and Raw2Digits for FDD togehter with CTF classes. All based on current FV0 implementation.

### CODE REVIEW

File: DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/LookUpTable.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      std::cout << std::right << std::setw(3) << mInvTopo[idx] << "    ";
+      std::cout << std::right << std::setw(2) << getLinkFromIdx(mInvTopo[idx]) << "  ";
+      std::cout << std::right << std::setw(2) << getModChannelFromIdx(mInvTopo[idx]) << std::endl;
+    }
+  }
+
+ private:
+  std::vector<Topo> mTopoVector; // iterator of each vector element gives the global channel number
+  std::vector<int> mInvTopo;     // each element is an iterator of mTopoVector
+  static constexpr int mLinkTCM = Nmodules;
```
Comment (shahor02) [inline DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/LookUpTable.h]: constants should be capitalized.

File: Detectors/FIT/FDD/simulation/src/Digits2Raw.cxx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+  std::string outd = outDir;
+  if (outd.back() != '/') {
+    outd += '/';
+  }
+
+  // Register PM links linearly
+  for (int iPmLink = 0; iPmLink < Nmodules; ++iPmLink) {
+    uint64_t feeId = uint64_t(iPmLink);
+    uint32_t linkId = uint32_t(iPmLink);
```
Comment (shahor02) [inline Detectors/FIT/FDD/simulation/src/Digits2Raw.cxx]: feeId is actually ``uint16_t`` and the linkId is ``uint8_t``, see 
``RawFileWriter::registerLink(uint16_t fee, uint16_t cru, uint8_t link, uint8_t endpoint, std::string_view outFileNameV)``

File: Detectors/FIT/FDD/simulation/src/Digits2Raw.cxx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    std::string outFileLink = mOutputPerLink ? (outd + "fdd_link" + std::to_string(iPmLink) + ".raw") : (outd + "fdd.raw");
+    LOG(INFO) << " Register PM link: " << iPmLink << " to file: " << outFileLink;
+    mWriter.registerLink(feeId, sCruId, linkId, sEndPointId, outFileLink);
+  }
+
+  // Register TCM link separately
+  std::string outFileLink = mOutputPerLink ? (outd + "fdd_link" + std::to_string(sTcmLink) + ".raw") : (outd + "fdd.raw");
+  LOG(INFO) << " Register TCM link: " << outFileLink;
+  mWriter.registerLink(uint64_t(sTcmLink), sCruId, sTcmLink, sEndPointId, outFileLink);
```
Comment (shahor02) [inline Detectors/FIT/FDD/simulation/src/Digits2Raw.cxx]: idem

**Review by shahor02 — CHANGES_REQUESTED** [review]
Hi @mbroz84 

Thanks! Looks fine, please see a few minor comments below. Could you please amend this PR by
1) a unit test for CTF encoding/decoding, you can mostly borrow the code from e.g. https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/CTF/test/test_ctf_io_fv0.cxx
and make sure that the original and decoding digits are the same.
2) add FT0 to https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/CTF/workflow/src/CTFReaderSpec.cxx, https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/CTF/workflow/src/CTFWriterSpec.cxx and https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/CTF/workflow/src/ctf-reader-workflow.cxx
Again, you need just to copy/paste the entries of e.g. FV0

**Comment by shahor02** [comment]
@mbroz84 just to ping that there your response is pending on this PR

**Review by AllaMaevskaya — DISMISSED** [review]

**Comment by mbroz84** [comment]
Sorry for delay, requested changes should be in the last commit.

**Comment by shahor02** [comment]
@mbroz84 Thanks! everything looks fine but the test_ctf_io_fdd you added fails on the macos, see the output in https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/4659/8eb75ba486c75be50e0f0f3b0dfa7f46fc89f522/build_AliceO2_O2_o2_macOS/fullLog.txt

Note that you can run the test yourself by going to build dir. and running like
````
shahoian@alicers02:~/alice/sw/BUILD/O2-latest/O2$ ./stage/tests/o2-test-ctf-fdd
````

**Review by shahor02 — APPROVED** [review]

**Pull request merged**